entries/git-workflow-guidelines/branching.md

```diff
@@ -0,0 +1,143 @@
+
+Hopefully, this gave you an intuition about what a branch is and what the benefits of branching are. If you would like more information on branches, check out the [GitHub Documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches) and the [Git Documentation Book](https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell).
+
+## When To Create Branches (and How Much They Should Cover)
+
+Every time you want to add something new to or modify existing work in a Git repository, you should create a new branch (following [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow)). Importantly, each branch should represent a specific task or development effort. I.e., a branch should be created for one particular purpose, independently of how many new lines of work is required for that purpose.
+
+If you create branches for *one* particular purpose, it benefits your (and your collaborators') workflow in (at least) two ways:
+
+1. It enables you to give the branch a short, descriptive name that clearly communicates to your collaborators what work has been done on this particular branch.
+2. Smaller changes allows for a faster reviewing and merging to the main branch, since it is easier for your collaborators to get an overview of the work you have done.
+
+As an example, a branch could cover a fix to a particular bug or [issue](issues.md) or add a new feature, such as a new plot, like in the figure showing parallel development above.
+
+## Branch Naming
+
+When you create a branch, it's important to give the branch a short, descriptive name to clearly communicate to your collaborators what this work is on. When naming a branch, stick to lowercase alphanumeric characters (a-z, 0-9) and use hyphens to separate words. Avoid punctuation, spaces, underscores, or other any non-alphanumeric characters than hyphen. We encourage the use of prefixes so anyone can easily identify what kind of work is being done on a certain branch. In addition, we recommend using a forward slash (/) after the prefix. The use of  slash results in the name being interpreted by many IDEs and Git tools as a directory structure, which creates a nice grouping of the branches. For prefixes and examples of branch names, see the table below.
+
+| Type | Use case | Pattern | Example |
+|-|-----|-----|-----|
+| Bugfix | For fixing a bug | bugfix/(short-description) | bugfix/fix-data-cleaning-error |
+| Feature | For adding, removing, or modifying a feature* | feature/(short-description) | feature/add-variance-plot |
+| Hotfix | For quickly fixing critical issues, usually with a temporary solution | hotfix/(short-description) | hotfix/fix-select-button |
+| Doc | For writing, updating, or fixing documentation | doc/(short-description) | doc/git-guidelines |
```

Comment:
I think we need more "Types" for non-code items, like the documentation and eventual training material. `doc/` is too broad.

Reply:
Agreed! I'll look into conventions for that

Reply:
@lwjohnst86 I found it difficult to find actual conventions for dividing doc into more types, but we could create our own? Something like this maybe: 

- docs-training/ or docs-learning/ (training materials) 
- docs-why/ (decision posts)
- docs-design/ (considerations on design) 
- docs-code/ (documentation on specific code?)  
- docs-blogs/ (for informal "learning"/something we have learned e.g. at a conference or meeting??)
- docs-misc/ (for things that doesn't fit any of the types we define - we can evaluate on this later to see if there are some "types" in here that we have missed) 

Do you have ideas?

Reply:
PS the VS Code extension [Conventional branch](https://marketplace.visualstudio.com/items?itemName=pshaddel.conventional-branch) makes it easy to create custom branch types to choose from - in a similar manner as Conventional Commits. Conventional branch is not widely used, though (391 installations)

Reply:
I'm happy to trial it if you guys feel like we should explore it.